kbchat/kbchat_test.go

```diff
@@ -103,82 +103,290 @@ func deleteWorkingDir(workingDir string) error {
 	return os.RemoveAll(workingDir)
 }
 
-var kbc *API
-var config Config
-
-func TestMain(m *testing.M) {
+func testSetup() (alice *API, config botConfig, dir string, oneOnOneChannel Channel, teamChannel Channel) {
 	var err error
 	config, err = readAndParseConfig()
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Error in reading config: %v\n", err)
 	}
-	dir, err := randomTempDir()
+	dir, err = randomTempDir()
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Error in generating directory: %v\n", err)
 	}
 	kbLocation, err := prepWorkingDir(dir)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Error in preparing working directory: %v\n", err)
 	}
-	kbc, err = Start(RunOptions{KeybaseLocation: kbLocation, HomeDir: dir, Oneshot: config.Bots.Alice1, StartService: true})
+	alice, err = Start(RunOptions{KeybaseLocation: kbLocation, HomeDir: dir, Oneshot: config.Bots["alice1"], StartService: true})
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Error in starting service: %v\n", err)
 	}
 
-	flag.Parse()
-	code := m.Run()
+	oneOnOneChannel = Channel{
+		Name: fmt.Sprintf("%s,%s", config.Bots["alice1"].Username, config.Bots["charlie1"].Username),
+	}
+	teamChannel = Channel{
+		Name:        config.Teams["acme"].Teamname,
```

Comment:
maybe also allow the caller to pass in the teamname that this is using, and default to this one?